src/main/java/org/folio/bulkops/processor/marc/MarcFlowCommitProcessor.java

```diff
@@ -0,0 +1,179 @@
+          }
+        }
+        if (nonNull(bulkOperation.getLinkToCommittedRecordsCsvFile())) {
+          var appendedCsvRecords = writer.toString();
+          appendedCsvRecords = appendedCsvRecords.substring(appendedCsvRecords.indexOf(LINE_BREAK) + 1);
+          var appendedCsvStream = new SequenceInputStream(
+            remoteFileSystemClient.get(bulkOperation.getLinkToCommittedRecordsCsvFile()),
+            new ByteArrayInputStream(appendedCsvRecords.getBytes()));
+          remoteFileSystemClient.put(appendedCsvStream, committedCsvFileName);
+        }
+      } catch (Exception e) {
+        log.error("Failed to enrich csv file", e);
+      } finally {
+        if (nonNull(bulkOperation.getLinkToCommittedRecordsCsvFile())) {
+          remoteFileSystemClient.remove(bulkOperation.getLinkToCommittedRecordsCsvFile());
+        }
+        bulkOperation.setLinkToCommittedRecordsCsvFile(committedCsvFileName);
+      }
+    }
+  }
+
+  private void writeBeanToCsv(BulkOperationsEntityCsvWriter writer, BulkOperationsEntity bean)
+    throws CsvRequiredFieldEmptyException, CsvDataTypeMismatchException {
+    try {
+      writer.write(bean);
+    } catch (ConverterException e) {
+      writeBeanToCsv(writer, bean);
+    }
+  }
+
+  public void enrichCommittedMarcWithUpdatedInventoryRecords(BulkOperation bulkOperation, List<String> csvHrids, List<String> marcHrids) {
+    var hrids = CollectionUtils.subtract(csvHrids, marcHrids);
+    if (!hrids.isEmpty() && nonNull(bulkOperation.getLinkToMatchedRecordsMarcFile())) {
+      var linkToCommitted = bulkOperation.getLinkToCommittedRecordsMarcFile();
+      var dirName = isNull(linkToCommitted) ? bulkOperation.getId().toString() : ENRICHED_PREFIX + bulkOperation.getId();
+      var committedMarcFileName = CHANGED_MARC_PATH_TEMPLATE.formatted(dirName, LocalDate.now(), getBaseName(bulkOperation.getLinkToTriggeringCsvFile()));
+      try (var matchedMarcInputStream = remoteFileSystemClient.get(bulkOperation.getLinkToMatchedRecordsMarcFile());
+           var committedMarcInputStream = isNull(linkToCommitted) ? null : remoteFileSystemClient.get(linkToCommitted);
+           var marcOutputStream = new ByteArrayOutputStream()) {
+        var marcReader = new MarcStreamReader(matchedMarcInputStream);
+        var streamWriter = new MarcStreamWriter(marcOutputStream, "UTF-8");
+        while (marcReader.hasNext()) {
+          var marcRecord = marcReader.next();
+          if (hrids.contains(marcRecord.getControlNumber())) {
+            streamWriter.write(marcRecord);
+          }
+        }
+        var appendedMarcStream = new ByteArrayInputStream(marcOutputStream.toString(UTF_8).getBytes());
+        var committedMarcStream = isNull(linkToCommitted) ?
+          appendedMarcStream :
+          new SequenceInputStream(committedMarcInputStream, appendedMarcStream);
+        remoteFileSystemClient.put(committedMarcStream, committedMarcFileName);
+      } catch (Exception e) {
```

Comment:
The same with exceptions handling and processing S3 exceptions like for FOLIO cases.